torch/_dynamo/utils.py

```diff
@@ -915,8 +835,30 @@ def add_compilation_metrics_to_chromium(c: CompilationMetrics):
     )
 
 
-def record_compilation_metrics(compilation_metrics: CompilationMetrics):
-    global _compilation_metrics
+def record_compilation_metrics(metrics: Dict[str, Any], exc_type, exc_value):
+    # Temporary: populate legacy fields from their replacements.
+    # Remove when we decide we can really deprecate them.
+    def legacy(field):
+        metric = metrics.get(field, None)
+        if not metric:
+            return None
+        return float(metric) / 1e6
```

Comment:
Instead of converting metric to float first, just do metric / 1e6 which is guaranteed to be float because int / float produces float. Because Python is nuts this will actually happen in higher precision than what you have here.